app/models/noise.rb

```diff
@@ -24,13 +24,76 @@ class Noise < ActiveRecord::Base
     using: {tsearch: {dictionary: "english"}},
     associated_against: { user: :name }
 
-  delegate :url_helpers, to: 'Rails.application.routes'
-
   belongs_to :user
   has_one :original, as: :importable, dependent: :destroy
   has_many :origins, ->{ uniq true },
     dependent: :destroy
 
+  scope :where_nearby, ->(params) do
+    location = params[:location]
+    latitude = params[:latitude]
+    longitude = params[:longitude]
+
+    if latitude && longitude
+      search_location = [latitude, longitude]
+    elsif location
+      search_location = location
+    end
+
+    if search_location
+      Rails.logger.debug "Location detected " + search_location.to_s
+
+      noise_ids = Origin.explore(params)
+        .where_since(params[:created_at])
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.